pm/recipient.go

```
@@ -332,16 +334,12 @@ func (r *recipient) redeemWinningTicket(ticket *Ticket, sig []byte, recipientRan
 	// If max float is insufficient to cover the ticket face value, queue
 	// the ticket to be retried later
 	if maxFloat.Cmp(ticket.FaceValue) < 0 {
-		if err := r.sm.QueueTicket(ticket.Sender, &SignedTicket{ticket, sig, recipientRand}); err != nil {
-			return err
-		}
-
+		r.sm.QueueTicket(ticket.Sender, &SignedTicket{ticket, sig, recipientRand})
 		glog.Infof("Queued ticket sender=%x recipientRandHash=%x senderNonce=%v", ticket.Sender, ticket.RecipientRandHash, ticket.SenderNonce)
-
 		return nil
 	}
 
-	info, err := r.broker.GetSenderInfo(ticket.Sender)
+	info, err := r.smgr.GetSenderInfo(ticket.Sender)
```

I think we can actually get rid of this and the check on L349 and instead before we check if `maxFloat < ticket.FaceValue` we can check if `maxFloat = 0`. If `maxFloat = 0` then we return an error. The reasoning is that if `maxFloat > 0` then the sender definitely has a non-zero reserve so the current ticket could be eligible for redemption. And if `maxFloat = 0` then either the sender's reserve is 0 OR the current ticket is not eligible for redemption because the recipient has already claimed the max amount from the sender's reserve. In either case, we can just return an error and not try to submit the ticket on-chain.

Removing this check would also remove the `SenderManager` dependency in `Recipient` which is an added bonus.